internal/sapsystem/sapsystem.go

```diff
@@ -233,23 +232,20 @@ func getProfilePath(sysPath string) string {
 }
 
 // Get SAP profile file content
-func getProfileData(fs afero.Fs, profilePath string) (map[string]interface{}, error) {
-	profile, err := fs.Open(profilePath)
+func getProfileData(fs afero.Fs, profilePath string) (map[string]string, error) {
+	profileFile, err := fs.Open(profilePath)
 	if err != nil {
 		return nil, errors.Wrap(err, "could not open profile file")
 	}
 
-	defer profile.Close()
-
-	profileRaw, err := io.ReadAll(profile)
+	defer profileFile.Close()
```

Comment:
Like

`defer func(){
err := profileFile.Close()
if err != nil {
  // log here with warning
}
}

Reply:
Some reference about that https://www.joeshaw.org/dont-defer-close-on-writable-files/, of course we are reading and not writing but the solutions proposed are used patterns in golang. I prefer in this case to just log it because it seems not affecting the general correctness of the flow

Reply:
Anyway I think the linter should catch that stuff, but maybe on defer it has some problems

Reply:
thanks i totally missed this

Reply:
well, not totally fair to say that you missed. It was this way already, so we missed long time ago XD

Reply:
I learned the hard way this closing thing lel, one day I will tell you a story about a faulty file closing and thousands of files not saved to disk 👀